src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -33,9 +34,15 @@ public CalculationResults GetSuggestedExpensesOffers(IEntryList entryList, IGoal
                 generateRandomData = true;              
             }
 
-            for (var i = enumCount; i > (int)Importance.Necessary; i--)
+            var listToUse = generateRandomData ? expensesRandomList : entryList;
+            var groupedByImportance = (from entry in listToUse
+                                       where entry.Importance != 1
+                                       group entry by entry.Importance).ToArray();
```

Comment:
Could you explain what does it return? (since we need to use `ToArray()` )

Reply:
It returns an IEnumerable of IGrouping, but since it is more efficient to traverse an array than an IEnumerable, we cast it to an array. Of course this is not really necessary, but since originally we thought maybe the list will be traversed multiple times this bit was added ( as suggested by resharper) :D